scene/gui/line_edit.cpp

```diff
@@ -569,22 +576,51 @@ bool LineEdit::can_drop_data(const Point2 &p_point, const Variant &p_data) const
 		return drop_override;
 	}
 
-	return p_data.get_type() == Variant::STRING;
+	return is_editable() && p_data.get_type() == Variant::STRING;
 }
 
 void LineEdit::drop_data(const Point2 &p_point, const Variant &p_data) {
 	Control::drop_data(p_point, p_data);
 
-	if (p_data.get_type() == Variant::STRING) {
+	if (is_editable() && p_data.get_type() == Variant::STRING) {
 		set_caret_at_pixel_pos(p_point.x);
-		int selected = selection.end - selection.begin;
-
-		text.erase(selection.begin, selected);
-		_shape();
+		int caret_column_tmp = caret_column;
+		if (selection.drag_attempt) {
+			selection.drag_attempt = false;
+			if (caret_column < selection.begin || caret_column > selection.end) {
+				if (caret_column_tmp > selection.end) {
+					caret_column_tmp = caret_column_tmp - (selection.end - selection.begin);
+				}
+				delete_text(selection.begin, selection.end);
+				deselect();
 
-		insert_text_at_caret(p_data);
-		selection.begin = caret_column - selected;
-		selection.end = caret_column;
+				set_caret_column(caret_column_tmp);
+				insert_text_at_caret(p_data);
+			}
+		} else if (selection.enabled && caret_column >= selection.begin && caret_column <= selection.end) {
+			caret_column_tmp = selection.begin;
+			delete_text(selection.begin, selection.end);
+			deselect();
+			set_caret_column(caret_column_tmp);
+			insert_text_at_caret(p_data);
+			grab_focus();
+		} else {
+			deselect();
```

Comment:
Can remove this call to `deselect` as we're immediately selecting afterwords.

Reply:
fixed